src/navigation/navigation-events.js

```diff
@@ -99,15 +99,21 @@ const originalPushState = window.history.pushState;
 window.history.pushState = function(state) {
   const result = originalPushState.apply(this, arguments);
 
-  reroute();
+  // https://github.com/CanopyTax/single-spa/issues/224 and https://github.com/CanopyTax/single-spa-angular/issues/49
+  // We need a popstate event even though the browser doesn't do one by default when you call pushState, so that
+  // all the applications can reroute.
+  urlReroute(new PopStateEvent('popstate'));
```

Comment:
I think the two options are two do it this way (which violates the browser spec for pushState) or alter all of the single-spa helper libraries to listen to single-spa's routing event listener and forcibly update when it happens

Reply:
That’s a good point, and may warrant a bigger discussion? 

I almost lean towards not doing things that are different than the spec, because it opens the door to future bugs when we’re firing that event and it shouldn’t be

Reply:
I agree -- lets add this to the next core team meeting  agenda?

I think i'll merge this in for now since. I really do think the chance of breaking people's app is pretty small ("what's one more re-render?" famous last words), but am open to reversing this decision and code later if we as a core team decided on a different approach.

Reply:
Bad news: I think the alternative implementation doesn't work, although I'm not sure why. Just tried it out with single-spa-angular